src/aa.select2.js

```diff
@@ -87,8 +87,16 @@ angular
           //run a query to get options (search)
           if (inAjaxMode) {
             derivedOpts.query = function (query) {
-              settings.options(query.term)
-                .success(function (data) {
+              // TODO: Ought to handle the failure case, instead of spinning forever
+              var queryResult = settings.options(query.term);
+              // Assume if it's not a promise and is an array, it's data
+              if (!queryResult.then && queryResult.hasOwnProperty["length"]) {
```

Comment:
Checking whether it's an array like object. Don't care if it's actually an array, as long as behaves like one.